src/nvim/option.c

```diff
@@ -1150,9 +1136,12 @@ do_set (
        */
       arg += 3;
       if (*arg == '&') {
-        ++arg;
-        /* Only for :set command set global value of local options. */
+        arg++;
+        // Only for :set command set global value of local options.
         set_options_default(OPT_FREE | opt_flags);
+        didset_options();
+        didset_options2();
+        redraw_all_later(CLEAR);
```

Comment:
Normally I'd try to see if we could avoid this, since we don't allow changing encoding--but `:set all&` is only used once per session, if at all, so it doesn't matter.